src/Servers/Kestrel/Core/src/Internal/Infrastructure/StringUtilities.cs

```diff
@@ -109,6 +112,127 @@ public static unsafe bool TryGetAsciiString(byte* input, char* output, int count
             return isValid;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        public unsafe static bool BytesOrdinalEqualsStringAndAscii(string previousValue, Span<byte> newValue)
```

Comment:
Are there plans to add anything to BCL for efficiently comparing a string to an ASCII-encoded span?

Reply:
Not currently when I asked. It would likely need to be a Utf8 to `string` compare to fit in with any current apis, however that's a lot more complicated. 

Not sure where a simpler ascii-to-string Ordinal only would fit to make an api proposal; as it would likely also need IgnoreCase and CultureAware variants to match with the kind of apis corefx would want; and then its suddenly a lot more complicated and a lot of code for probably a niche case.

/cc @GrabYourPitchforks

Reply:
We did just introduce a type `Utf8` into the `System.Text.Unicode` namespace, and we have plans to add APIs like `ToUpper` to that type. Maybe we'd also put an `Ascii` type in that same namespace? (Yuck.)

Reply:
Only want something equivalent to `string.CompareOrdinal(ReadOnlySpan<byte> asciiBytes)`

Could also go with `Utf8.CompareOrdinal(string, ReadOnlySpan<byte>)` if it was fast in the ascii-only path; Utf8 headers should be rare and is little value in optimizing for them; although I assume corefx would care about the performance of the Utf8 path also since it would be a more generic method.

Reply:
@GrabYourPitchforks can we find a home for this? It would be great if this didn't have to live in Kestrel 😄

Reply:
Open an issue in the corefx repo with what you want the API to be. :)

In general I'm pessimistic about a general-purpose API that compares UTF-8 and UTF-16 directly, as the performance of such a thing would be abysmal except for all-ASCII input. And if you're looking for an _ASCII-specific_ API, definitely call out that specific scenario when you open the issue.

Reply:
@benaadams can you file an API request?

Reply:
https://github.com/dotnet/corefx/issues/36519